### PR TITLE
Fix argMin/argMax combinator state

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Fix argMin/argMax combinator state

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 1c0d104b482e01e9a99d6c1a92cf4f1c34cec51f with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62708/1c0d104b482e01e9a99d6c1a92cf4f1c34cec51f/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/queries/0_stateless/03127_argMin_combinator_state.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        avgArgMaxState(CAST(number, 'Float64'), id)
+    FROM numbers(100)
+    GROUP BY id;
+SELECT avgArgMaxMerge(state) FROM argmax_comb;
+SELECT
+    id,
+    avgArgMaxMerge(state)
+FROM argmax_comb
+GROUP BY id
+ORDER BY id ASC;
```
Comment (amosbird) [inline tests/queries/0_stateless/03127_argMin_combinator_state.sql:22]: Missing EOL. Consider using an editor plugin to do this chore :)

**Review by antonio2368 — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Comment by Algunenano** [comment]
The fuzzer found a problem that needs to be investigated using the new tests as a baseline

**Comment by Algunenano** [comment]
It took a while to extract the settings from fuzzer, but here is the repro:
```
SET allow_suspicious_low_cardinality_types = 1;
SET max_rows_to_group_by = 65535;
SET max_threads = 1;
SET max_block_size = 65536;
SET group_by_overflow_mode = 'any';
SET totals_mode = 'after_having_auto';
SET totals_auto_threshold = 0.5;

CREATE TABLE combinator_argMin_table_r1__fuzz_791
(
    `id` LowCardinality(Int32),
    `value`    String,
    `agg_time` UInt64,
    PROJECTION first_items (SELECT id, minArgMin(agg_time, value), maxArgMax(agg_time, value)
                            GROUP BY id)
)
    ENGINE = MergeTree ORDER BY id;

INSERT INTO combinator_argMin_table_r1__fuzz_791
SELECT number % 10 AS id,
       number      AS value,
       '01-01-2024 00:00:00' + toIntervalDay(number)
FROM numbers(100);

SELECT maxArgMax(agg_time, value)
FROM combinator_argMin_table_r1__fuzz_791
GROUP BY id WITH CUBE WITH TOTALS;
```

It seems WITH TOTALS is passing a state to be merged that's empty (no data has been added to it). It should not be related to the changes, but I'm still investigating.

**Pull request merged**